src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1004,7 +1009,32 @@ internal static IEnumerable<EnvDTE.Project> GetDependentEnvDTEProjects(IDictiona
             dependentEnvDTEProjects.Add(dependentEnvDTEProject);
         }
 
-#region IVsSelectionEvents
+        private void NuGetCacheUpdate_After(object sender, EventArgs e)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => FireNuGetCahceUpdatedEventAsync(e));
+        }
+
+        private async Task FireNuGetCahceUpdatedEventAsync(EventArgs e)
+        {
+            try
+            {
+                // Await a delay of 100 mSec to batch multiple cache updated events
+                await Task.Delay(100);
+                // Check if the cache is still dirty
+                if (_projectSystemCache.TestResetDirtyFlag())
```

Comment:
This check seems like it would be very cheap, should this also be done before starting a new thread?